docs/FiBo_RUP_Main_Test_Plan.md

```diff
@@ -0,0 +1,496 @@
+| Technique Objective: | Exercise the installation of the target-of-test onto each required hardware configuration under the following conditions to observe and log installation behavior and configuration state changes |
+| Technique: | Manual performed installation |
+| Oracles: | The app can be installed and used on Android devices |
+| Required Tools: | Android > 8|
+| Success Criteria: | installation successful |
+| Special Considerations: | - |
+
+### 6. Entry and Exit Criteria (Markus)
+### 6.1     Test Plan (Markus)
+#### 6.1.1     Test Plan Entry Criteria
+Building a new version of the software will execute the test process.
+
+#### 6.1.2     Test Plan Exit Criteria
+When all tests pass without throwing an exception.
+
+#### 6.1.3     Suspension and Resumption Criteria
+The test plan is not supposed to be suspended. Therefore, there are no suspension and resumption criteria.
+
+### 6.2      Test Cycles (Markus)
+#### 6.2.1     Test Cycle Entry Criteria
+A test cycle begins when the build of a new version is initiated.
+
+#### 6.2.2     Test Cycle Exit Criteria
+The test effort of a cycle is deemed sufficient when a build has been completed without failures and all unit tests are run successfully.
+
+#### 6.2.3     Test Cycle Abnormal Termination
+Occurring errors during the build terminate the test cycle.
+  
+### 7.     Deliverables (Markus)
+
+#### 7.1     Test Evaluation Summaries
+Test evaluation will be available online every time a Pull Request is merged into the main branch. The report is generated with the help of [CodeCov](https://codecov.io/gh/Cebox82/FiBo).
+#### 7.2     Reporting on Test Coverage
+After building and testing the master branch the resulting coverage is uploaded to [CodeCov](https://codecov.io/gh/Cebox82/FiBo).
+[![code coverage score](https://codecov.io/gh/Cebox82/FiBo/branch/master/graph/badge.svg)](https://codecov.io/gh/Cebox82/FiBo)
+#### 7.3     Perceived Quality Reports
+Code quality will be measured with SonarCloud. It will create a report for every pull request. The quality report status is showed by the following badge:
+[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=Cebox82_FiBo&metric=alert_status)](https://sonarcloud.io/summary/new_code?id=Cebox82_FiBo)
+
+
+#### 7.4     Incident Logs and Change Requests
+(n/a) `weglassen?`
+#### 7.5     Smoke Test Suite and Supporting Test Scripts
+(n/a) `weglassen?`
+
+
+#### 7.6      Additional Work Products
+##### 7.6.1     Detailed Test Results
+The following screenshot shows the unit test results after running with Android Studio:
+![](https://pad.medialepfade.net/uploads/d646ea36-b2de-461d-9b81-cff2241c02cb.png)
+
+This screenshot shows the results of running the UI tests:
+![](https://pad.medialepfade.net/uploads/f20c290a-a21d-48ef-84c2-a561de453ac5.png)
+
+This screenshot shows the results of running the Backend Django tests:
+![](https://pad.medialepfade.net/uploads/24f957c4-7f43-4cae-998e-08c653645b38.png)
```

Comment:
These Pictures should be uploaded to Github. We should not use the mediapad links, because they may become invalid